pkg/apis/core/types.go

```diff
@@ -2053,6 +2080,12 @@ type Container struct {
 	// Compute resource requirements.
 	// +optional
 	Resources ResourceRequirements
+	// Node compute resources allocated to the container.
+	// +optional
+	ResourcesAllocated ResourceList
```

Comment:
Why is this `spec` and not `status` ?

Reply:
This field was added only so that we have a source-of-truth of actual node resource allocations to the containers and the pod, should the status field be erased or lost - API conventions need status to be reconstructible.

The status field now has a new Resources field (type ResourceRequirements) to reflect that actual config reported by the runtime via extended CRI

Reply:
Wouldn't it be reconstructible from the Kubelet?  Being in spec requires all the admission control mess which is a really string smell that something is off, IMO

Reply:
Short answer - no.

For point-in-time, there is no way to discover memory request today - it is not stored anywhere. We can reconstruct current CPU requests, limits & memory limits from the extended ContainerStatus CRI API (not today) once runtime supports the CRI extension implemented in this PR.

We had discussed the alternative of checkpointing this locally on the node via atomic file-writes but Dawn mentioned they had run into issues with that kind of approach in Borg, and felt it was safest to have PodSpec be the single source-of-truth.

We had a long discussion on this in sig-node meetings before finally settling on adding ResourcesAllocated to PodSpec, and bringing it up during the Live API review last year. For gory details and kubelet restart scenarios, please see https://github.com/kubernetes/enhancements/pull/686#discussion_r297384981